From 2a14a9247dbf8560a168d1017e7322e996eced3e Mon Sep 17 00:00:00 2001
From: zjin6 <zhuojun.jin@intel.com>
Date: Tue, 21 Jan 2014 09:10:54 +0800
Subject: [PATCH] Enable memory layout patch back

SELinux renders the existing memory layout patch vain. This patch fixed this issue.
It is designed for both 32 bits and 64 bits kernel.

Change-Id: I9dc50283e79e23c4bae1b68ab22a9d1ec31c14f5
Category: feature differentiation
Domain: AOSP.Houdini-Integration
Origin: internal
Signed-off-by: zjin6 <zhuojun.jin@intel.com>

diff --git a/init/init.c b/init/init.c
index ae96440..33956df 100644
--- a/init/init.c
+++ b/init/init.c
@@ -31,7 +31,6 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/personality.h>
 
 #include <selinux/selinux.h>
 #include <selinux/label.h>
@@ -917,19 +916,6 @@ static int property_service_init_action(int nargs, char **args)
     return 0;
 }
 
-static int personality_init_action(int nargs, char **args)
-{
-    char tmp[PROP_VALUE_MAX];
-    int ret;
-    ret = property_get("ro.config.personality", tmp);
-    if (ret && !strcmp(tmp, "compat_layout")) {
-        int old_personality;
-        old_personality = personality((unsigned long)-1);
-        personality(old_personality | ADDR_COMPAT_LAYOUT);
-    }
-    return 0;
-}
-
 static int signal_init_action(int nargs, char **args)
 {
     signal_init();
@@ -1241,7 +1227,6 @@ int main(int argc, char **argv)
     queue_builtin_action(mix_hwrng_into_linux_rng_action, "mix_hwrng_into_linux_rng");
 
     queue_builtin_action(property_service_init_action, "property_service_init");
-    queue_builtin_action(personality_init_action, "personality_init");
     queue_builtin_action(signal_init_action, "signal_init");
     queue_builtin_action(check_startup_action, "check_startup");
 
-- 
2.5.0

